### PR TITLE
[MHV-55400]: medications shipped tracking

### PR DESCRIPTION
**Pull request by dcloud**
## Summary

> As someone who receives my prescription refills by mail, I want to see when those medications were shipped.

- Add "Last refill shipped on " to `/my-health/medications/refill`, with a truck icon
- Add unit test

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-55400 (Add shipped meds to new refill page)

## Related PR

- department-of-veterans-affairs/vets-api/pull/16056

## Testing done

- Before: no "Last refill shipped on" appeared under the "...isn’t ready to refill"
- After: "Last refill shipped on" appears for items shipped in last few weeks
- Mocked data and viewed page
- Wrote unit test

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    Screen Shot 2024-03-27 at 14 37 12    | Screen Shot 2024-03-27 at 14 33 57 |
| Desktop | Screen Shot 2024-03-27 at 14 37 21  |  Screen Shot 2024-03-27 at 14 33 35    |

## What areas of the site does it impact?

MHV Medications

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/medications/components/PrescriptionDetails/VaPrescription.jsx [hunk 1]
```diff
@@ -166,7 +166,7 @@ const VaPrescription = prescription => {
                     className="vads-u-margin--0 vads-u-margin-bottom--1"
                     data-testid="shipped-on"
                   >
-                    {dateFormat(shippedOn?.[i]?.completeDateTime)}
+                    {dateFormat(latestTrackingStatus?.completeDateTime)}
```
Comment (agravell047) [inline src/applications/mhv/medications/components/PrescriptionDetails/VaPrescription.jsx:169]: are we supposed to be using the index here?

File: src/applications/mhv/medications/tests/e2e/fixtures/listOfPrescriptions.json [hunk 2]
```diff
@@ -854,13 +854,13 @@
...
-                "refillDate": "2023-09-23T04:00:00.000Z",
+                "refillDate": "September 24, 2023",
                 "refillRemaining": 1,
                 "facilityName": "SLC10 TEST LAB",
                 "orderedDate": "2023-08-27T04:00:00.000Z",
                 "quantity": 1,
                 "expirationDate": "2024-08-27T04:00:00.000Z",
-                "dispensedDate": "2023-08-29T04:00:00.000Z",
+                "dispensedDate": "September 24, 2023",
                 "stationNumber": "979",
```
Comment (agravell047) [inline src/applications/mhv/medications/tests/e2e/fixtures/listOfPrescriptions.json:864]: what was the reason for the format change?

Comment (RajiVenkatesh18) [reply src/applications/mhv/medications/tests/e2e/fixtures/listOfPrescriptions.json:864]: Time does not show in the UI and I used a fixture file that has tracking info for the refill history.

**Review by vmenshutin-bylight — DISMISSED** [review]

File: src/applications/mhv/medications/components/PrescriptionDetails/VaPrescription.jsx (hunk 1 above)
Comment (dcloud) [reply src/applications/mhv/medications/components/PrescriptionDetails/VaPrescription.jsx:169]: The code change on line 25, `const latestTrackingStatus = prescription?.trackingList?.[0];` pulls an object, the first item out of the trackingList array. 

We [updated the vets-api serialization](https://github.com/department-of-veterans-affairs/vets-api/pull/16056) to remove the unnecessary levels of nesting, so the API will serve a 1-level array.

**Review by agravell047 — DISMISSED** [review]

File: src/applications/mhv/medications/tests/e2e/fixtures/listOfPrescriptions.json (hunk 2 above)
Comment (dcloud) [reply src/applications/mhv/medications/tests/e2e/fixtures/listOfPrescriptions.json:864]: Do we actually expect the API to return a date in that format? I'd expect a a consistent format from the backend APIs...

Comment (RajiVenkatesh18) [reply src/applications/mhv/medications/tests/e2e/fixtures/listOfPrescriptions.json:864]: Made the suggested changes and pushed the updated one.

**Review by vmenshutin-bylight — APPROVED** [review]

**Pull request merged**